libcodechecker/server/client_db_access_server.py

```diff
@@ -124,31 +142,121 @@ def check_auth_in_request(self):
 
         return success
 
+    def __get_product_name(self):
+        """
+        Get product name from the request's URI.
+        """
+
+        # A standard request from a browser looks like:
+        # http://localhost:8001/[product-name]/#{request-parts}
+        # where the parts are, e.g.: run=[run_id]&report=[report_id]
+        #
+        # Rewrite the "product-name" so that the web-server deploys the
+        # viewer client from the www/ folder.
+
+        # The split array looks like ['', 'product-name', ...].
+        first_part = urlparse.urlparse(self.path).path.split('/', 2)[1]
+        return first_part if first_part not in NON_PRODUCT_NAMES else None
```

Comment:
Wouldn't be better to have an URL like `http://localhost:8001/product/[product-name]/#{request-parts}` so we do not need to maintain a list like `NON_PRODUCT_NAMES`?